submissions/nadsatt/dom-api/app.js

```diff
@@ -0,0 +1,192 @@
+        rating: 2,
+        colors: ['rgb(207,163,179)', 'rgb(146,43,62)', 'rgb(23,69,111)', 'rgb(189,200,196)']
+    },
+    {
+        id: 3,
+        name: 'kitty potter', 
+        description: 'Harry Potter\'s third year at Hogwarts starts off badly when he learns deranged killer Sirius Black has escaped from Azkaban prison and is bent on murdering the teenage wizard. While Hermione\'s (Emma Watson) cat torments Ron\'s sickly rat, causing a rift among the trio, a swarm of nasty Dementors is sent to protect the school from Black. A mysterious new teacher helps Harry learn to defend himself, but what is his secret tie to Sirius Black?',
+        rating: 9,
+        colors: ['rgb(0,145,194)', 'rgb(0,26,57)', 'rgb(175, 171, 171)', 'rgb(116,153,156)']
+    },
+    {
+        id: 4,
+        name: 'new meov', 
+        description: 'High-school student Bella Swan, always a bit of a misfit, doesn\'t expect life to change much when she moves from sunny Arizona to rainy Washington state. Then she meets Edward Cullen, a handsome but mysterious teen whose eyes seem to peer directly into her soul. Edward is a vampire whose family does not drink blood, and Bella, far from being frightened, enters into a dangerous romance with her immortal soulmate.',
+        rating: 5,
+        colors: ['rgb(23,131,72)', 'rgb(93, 150, 18)', 'rgb(175, 171, 171)', 'rgb(128,126,44)']
+    },
+];
+
+headerIcon.onclick = toggleSidebar;
+sidebarIcon.onclick = toggleSidebar;
+
+fillPageByMovieId(0);
+
+function toggleSidebar(){
+    headerIcon.classList.toggle('icon-wrapper--active');
+    sidebarIcon.classList.toggle('icon-wrapper--active');
+    sidebar.classList.toggle('sidebar--active');
+}
+
+function fillPageByMovieId(id){
+    const movie = movies.find(movie => id === movie.id);
+
+    fillSidebarList(movies.map(movie => movie.name));
+    fillPosterSection(movies[id].name);
+    fillRatingSection(movies[id].rating);
+    fillActorsSection(movies[id].name);
+    fillDescriptionSection(movies[id].description);
+    defineColors(movies[id].colors);
+    defineBackgrounds(movies[id].name);
+}
+
+function fillSidebarList(names){
+    if(sidebarList.firstElementChild){
+        const sidebarItems = sidebar.querySelectorAll('.sidebar__item');
+        sidebarItems.forEach((item, i) => item.textContent = names[i]);
+    }
+    else {
+        names.forEach(name => {
+            let sidebarItem = document.createElement('li');
+            sidebarItem.textContent = name;
+            sidebarItem.className = 'sidebar__item';
+
+            sidebarItem.onclick = function(){
+                let id = movies.find(movie => movie.name === this.textContent).id;
+                fillPageByMovieId(id)
```

Comment:
After each click, you go through all the steps again.
Think about how you can avoid this and relieve this problem.